### PR TITLE
Enable float type hyperparameters in WITH clause

### LINKED ISSUE #910: Attribute supports more type derivation
This issue comes from  PR: https://github.com/sql-machine-learning/models/pull/19/files, the WITH clause: `WITH model. pretrain_lr =0.01` should be converted to FLOAT type instead of STRING type, that we don't need to cast the type in the model constructor.

### PR DESCRIPTION
**Pull request by shendiaomo**
Fix #910

### CODE REVIEW

**Comment by weiguoz** [comment]
LGTM.

Few suggestions,
- Let's fix the error message.
https://github.com/sql-machine-learning/sqlflow/blob/2fa7a67db83b6ec236d343b3946d0231711e9754/sql/attribute.go#L149

- SQLFlow supports `[]int`
https://github.com/sql-machine-learning/sqlflow/blob/2fa7a67db83b6ec236d343b3946d0231711e9754/sql/attribute.go#L142
I'm not sure if we should support the `[]float32`. How do you think @typhoonzero

**Review by weiguoz — DISMISSED** [review]
LGTM

**Comment by shendiaomo** [comment]
> LGTM.
> 
> Few suggestions,
> 
> * Let's fix the error message.
>   https://github.com/sql-machine-learning/sqlflow/blob/2fa7a67db83b6ec236d343b3946d0231711e9754/sql/attribute.go#L149

Done.
> * SQLFlow supports `[]int`
>   https://github.com/sql-machine-learning/sqlflow/blob/2fa7a67db83b6ec236d343b3946d0231711e9754/sql/attribute.go#L142
>   
>   I'm not sure if we should support the `[]float32`. How do you think @typhoonzero

IMO float32 lists are rarely used in machine learning, but some models like libFM use lists of float32s to set regularizers for different model parts. Maybe we should consider this only when needed in the future.

File: sql/attribute.go [hunk 1]
```diff
@@ -146,7 +151,7 @@ func (a *attribute) GenerateCode() (string, error) {
 		return fmt.Sprintf("%s=%s", a.Name,
 			strings.Join(strings.Split(fmt.Sprint(intList), " "), ",")), nil
 	}
-	return "", fmt.Errorf("value of attribute must be string or list of int, given %s", a.Value)
+	return "", fmt.Errorf("The Value type of an attribute must be string, int, float or list of ints, given %s", a.Value)
```
Comment (typhoonzero) [inline sql/attribute.go]: Error string should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Review by typhoonzero — COMMENTED** [review]
#865 also does this and will be used if IR work is finished in the future, FYI.

File: sql/attribute.go (hunk 1 above)
Comment (shendiaomo) [reply sql/attribute.go]: Done.

**Review by weiguoz — APPROVED** [review]

**Review by typhoonzero — APPROVED** [review]

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Review by wangkuiyi — APPROVED** [review]
LGTM

**Pull request merged**